docs/adrs/010-logs.md

```diff
@@ -0,0 +1,35 @@
+# ADR 010: What to do with logs?
+
+**Problem:** when commands fail, you usually want to look at the process logs to see what went wrong.
+You also might want to see logs as the system is building.
+This is more complicated than just "ok, so stream them" because we plan to add remote builders.
+How do we persist logs so you can look at them later, for example when a CI job fails?
+
+It seems like the best solution would be to have both a log store and a log sink.
+Specifically:
+
+1. during evaluation of jobs, logs can be streamed.
+   This would let us show them in the CLI or in a web UI.
+2. Logs could be stored like job artifacts in the content-addressable store.
+
+Part 1 can be implemented today, but part two needs a little bit more groundwork.
```

Comment:
If you were to do this directly, wouldn't you just get a mess of mixed logs due to concurrent jobs?

Reply:
I'm not sure what you mean, here. We wouldn't be interleaving them—they'd be stored separately, but associated with an invocation.

Reply:
I specifically meant showing stdout and stderr in the CLI. With concurrent jobs theoretically run on a cluster being streaming back to a dev machine, I feel like it would generate an unreadable mess of logs.